src/waveform/renderers/waveformmark.h

```diff
@@ -8,46 +8,42 @@
 #include "util/memory.h"
 
 #include "waveform/renderers/waveformmarkproperties.h"
+#include "control/controlobject.h"
 
 class SkinContext;
 class WaveformSignalColors;
 
+class WOverview;
+
 class WaveformMark {
   public:
     static const int kDefaultHotCue = -1;
+    WaveformMark(const QString& group,
+                               const QDomNode& node,
+                               const SkinContext& context,
+                               const WaveformSignalColors& signalColors,
+                               int hotCue = kDefaultHotCue, 
+                               QString item = "");
 
-    explicit WaveformMark(int hotCue = kDefaultHotCue);
 
     // Disable copying
     WaveformMark(const WaveformMark&) = delete;
     WaveformMark& operator=(const WaveformMark&) = delete;
 
-    // Enable swapping
-    void swap(WaveformMark& that) {
-        std::swap(m_pPointCos, that.m_pPointCos);
-        std::swap(m_properties, that.m_properties);
-        std::swap(m_iHotCue, that.m_iHotCue);
-        std::swap(m_image, that.m_image);
-    }
-    friend void swap(WaveformMark& lhs, WaveformMark& rhs) {
-        lhs.swap(rhs);
-    }
-
-    void reset(int hotCue = kDefaultHotCue);
-
-    void setup(const QString& group, const QDomNode& node,
-               const SkinContext& context,
-               const WaveformSignalColors& signalColors);
-
-    std::unique_ptr<ControlProxy> m_pPointCos;
-
     const WaveformMarkProperties& getProperties() const { return m_properties; };
     void setProperties(const WaveformMarkProperties& properties) { m_properties = properties; };
 
     int getHotCue() const { return m_iHotCue; };
     void setHotCue(int hotCue) { m_iHotCue = hotCue; };
 
+    //The m_pPointCos related function
+    bool isValid() const{ return !(m_pPointCos == nullptr) && m_pPointCos->valid(); }
```

Comment:
Please insert a space before the curly brace. I prefer start the inlined code after a line break, especially when the lines get longish. The code will be shortened anyway, so you might leave it this way ;) But please be consistent with the spacing to improve readability.